tests/unit/test_charm.py

```diff
@@ -52,34 +78,28 @@ def test_many_relations(harness):
             "password": "",
         },
     )
-    rel_id1 = harness.add_relation("grpc", "grpc-one")
-    harness.add_relation_unit(rel_id1, "grpc-one/0")
+
+
+def setup_ingress_relation(harness: Harness):
+    rel_id = harness.add_relation("ingress", "istio-pilot")
+    harness.add_relation_unit(rel_id, "istio-pilot/0")
     harness.update_relation_data(
-        rel_id1,
-        "grpc-one",
-        {
-            "_supported_versions": "- v1",
-            "data": yaml.dump({"service": "grpc-one", "port": "8080"}),
-        },
+        rel_id,
+        "istio-pilot",
+        {"_supported_versions": "- v1"},
     )
+    return rel_id
 
-    rel_id2 = harness.add_relation("grpc", "grpc-two")
-    harness.add_relation_unit(rel_id2, "grpc-two/0")
+
+def setup_grpc_relation(harness: Harness, name: str, port: str):
```

Comment:
Dont we want to test the GRPC endpoint also in the integration test to actually do some grpc call ?

Reply:
hmm I 'm not sure about this since:
- Envoy operator is adding support for calls using gRPC-Web protocol
- Making an actual grpc call should be part of integration testing the mlmd component (grpc server)

Thus what we could add to integration tests is making an actual call using gRPC-Web protocol but that would require a client that could make those requests.

Reply:
hmm maybe https://github.com/public/sonora